packages/samples/headless-commerce-ssr-remix/app/root.tsx

```diff
@@ -0,0 +1,83 @@
+import externalCartAPI from '@/client/external-cart-api';
+import {json, LoaderFunctionArgs, MetaFunction} from '@remix-run/node';
+import {
+  Links,
+  Meta,
+  NavLink,
+  Outlet,
+  Scripts,
+  ScrollRestoration,
+  useLoaderData,
+} from '@remix-run/react';
+import {randomUUID} from 'crypto';
+import {coveo_visitorId} from './cookies.server';
+
+export const meta: MetaFunction = () => {
+  return [
+    {title: 'Coveo Commerce SSR + Remix'},
+    {name: 'description', content: 'Coveo Headless Commerce React SSR + Remix'},
+  ];
+};
+
+export const loader = async ({request}: LoaderFunctionArgs) => {
+  const totalItemsInCart = await externalCartAPI.getTotalCount();
+  const cookie = await coveo_visitorId.parse(request.headers.get('Cookie'));
+
+  if (!cookie) {
+    const visitorId = randomUUID();
+    return json(
+      {totalItemsInCart},
+      {
+        headers: {
+          'Set-Cookie': await coveo_visitorId.serialize(visitorId, {
+            encode: (value) => atob(value).replaceAll('"', ''),
+            maxAge: 60 * 24 * 365,
+          }),
+        },
+      }
+    );
+  }
+
+  return {totalItemsInCart};
+};
+
+export function Layout({children}: {children: React.ReactNode}) {
+  const {totalItemsInCart} = useLoaderData<typeof loader>();
+  const routes = [
+    {to: '/search', name: 'Search'},
+    {to: '/listings/surf-accessories', name: 'Surf Accessories'},
+    {to: '/listings/paddleboards', name: 'Paddleboards'},
+    {to: '/listings/toys', name: 'Toys'},
+  ];
+
+  return (
+    <html lang="en">
+      <head>
+        <meta charSet="utf-8" />
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+        <Meta />
+        <Links />
+      </head>
+      <body>
+        <h1>Coveo Commerce SSR + Remix</h1>
+        <div style={{display: 'flex', alignItems: 'center', gap: '10px'}}>
+          {routes.map((route) => (
+            <NavLink key={route.to} to={route.to} prefetch="intent">
```

Comment:
I think the Remix prefetch feature is pretty neat so I'm enabling it here.

Reply:
That might do unnecessary queries, no?

Reply:
Correct... and on second thought, given that we're logging server-side events perhaps it would be best to explicitly mention that this feature, however neat, should not be enabled for Coveo-powered pages :(